src/main/java/org/jenkinsci/plugins/github_branch_source/HttpsRepositoryUriResolver.java

```diff
@@ -31,7 +31,10 @@
 
     @Override
     public String getRepositoryUri(String apiUri, String owner, String repository) {
-        return "https://" + hostnameFromApiUri(apiUri) + "/" + owner + "/" + repository + ".git";
+        if (apiUri.startsWith("https://")) {
```

Comment:
:bug: IIUC this is going to be `""` or `null` in the normal case of public GitHub, thus leading to incorrect URIs in the normal checkout case.